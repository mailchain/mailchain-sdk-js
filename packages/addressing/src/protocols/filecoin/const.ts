import { SECP256K1PrivateKey } from '@mailchain/crypto';
import { decodeHex } from '@mailchain/encoding';

// Alice
export const AliceFilStr = 'f410faiikgixcdd7alrsbpunjlen4etkcarbfbghn7wi';
export const AliceFilEthAddressStr = '0x0210a322e218fe05c6417d1a9591bc24d4204425';
export const AliceFilEthPrivateKeyStr = '6c115551accbd74b46401a531af1dd0890034c18137dddab52a53c626e75b051';
export const AliceFilEthPrivateKey = new SECP256K1PrivateKey(decodeHex(AliceFilEthPrivateKeyStr));

// Bob
export const BobFilAddressStr = 'f410f5blfdi7tmto6o2aa2up4vwtdm7qcnaiqpeazuwy';
export const BobFilEthAddressStr = '0xe85651a3f364dde76800d51fcada6367e0268110';
export const BobFilEthPrivateKeyStr = '2c6b29d5545e5804b7fb7e65404782e57dddea84cbcec9b2586add6403808965';
export const BobFilEthPrivateKey = new SECP256K1PrivateKey(decodeHex(BobFilEthPrivateKeyStr));

export enum FilPrefix {
	F4ETHEREUM = 'f410',
	T4ETHEREUM = 't410',
}
