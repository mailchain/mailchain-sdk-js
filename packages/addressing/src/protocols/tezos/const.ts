export enum Prefix {
	TZ1 = 'tz1',
	TZ2 = 'tz2',
	TZ3 = 'tz3',
	KT = 'KT',
	KT1 = 'KT1',

	EDSK2 = 'edsk2',
	SPSK = 'spsk',
	P2SK = 'p2sk',

	EDPK = 'edpk',
	SPPK = 'sppk',
	P2PK = 'p2pk',

	EDESK = 'edesk',
	SPESK = 'spesk',
	P2ESK = 'p2esk',

	EDSK = 'edsk',
	EDSIG = 'edsig',
	SPSIG = 'spsig',
	P2SIG = 'p2sig',
	SIG = 'sig',
	EXPR = 'expr',
}

export const prefix = {
	[Prefix.TZ1]: new Uint8Array([6, 161, 159]),
	[Prefix.TZ2]: new Uint8Array([6, 161, 161]),
	[Prefix.TZ3]: new Uint8Array([6, 161, 164]),
	[Prefix.KT]: new Uint8Array([2, 90, 121]),
	[Prefix.KT1]: new Uint8Array([2, 90, 121]),

	[Prefix.EDSK]: new Uint8Array([43, 246, 78, 7]),
	[Prefix.EDSK2]: new Uint8Array([13, 15, 58, 7]),
	[Prefix.SPSK]: new Uint8Array([17, 162, 224, 201]),
	[Prefix.P2SK]: new Uint8Array([16, 81, 238, 189]),

	[Prefix.EDPK]: new Uint8Array([13, 15, 37, 217]),
	[Prefix.SPPK]: new Uint8Array([3, 254, 226, 86]),
	[Prefix.P2PK]: new Uint8Array([3, 178, 139, 127]),

	[Prefix.EDESK]: new Uint8Array([7, 90, 60, 179, 41]),
	[Prefix.SPESK]: new Uint8Array([0x09, 0xed, 0xf1, 0xae, 0x96]),
	[Prefix.P2ESK]: new Uint8Array([0x09, 0x30, 0x39, 0x73, 0xab]),

	[Prefix.EDSIG]: new Uint8Array([9, 245, 205, 134, 18]),
	[Prefix.SPSIG]: new Uint8Array([13, 115, 101, 19, 63]),
	[Prefix.P2SIG]: new Uint8Array([54, 240, 44, 52]),
	[Prefix.SIG]: new Uint8Array([4, 130, 43]),
	[Prefix.EXPR]: new Uint8Array([13, 44, 64, 27]),
};
