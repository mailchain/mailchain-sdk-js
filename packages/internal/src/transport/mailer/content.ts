import { publicKeyFromBytes, publicKeyToBytes, PublicKey } from '@mailchain/crypto';
import { decodeHex, encodeHex } from '@mailchain/encoding';
import { MailerProof } from '@mailchain/signatures';
import canonicalize from 'canonicalize';
import { MailAddress } from '../mail';

/**
 * MailerContent is the content that is sent by the Mailer to each recipient.
 */
export type MailerContent = {
	/**
	 * The author's messaging public key, this MUST match the messaging key in the final `from` address.
	 */
	authorMessagingKey: PublicKey;
	/**
	 * Where the content created by the author can be found.
	 */
	contentUri: string;
	/**
	 * Date message is sent, this will become the Date: header in the email.
	 */
	date: Date;
	/**
	 * The author's mail address, this must resolve to the author messaging key. The will become the `From:` header in the email.
	 */
	authorMailAddress: MailAddress;
	/**
	 * Proof created by the author that authorizes the Mailer to send the message.
	 */
	mailerProof: MailerProof;
	/**
	 * The message ID, this will become the Message-ID: header in the email.
	 * This MUST be unique for each message.
	 */
	messageId: string;
	/**
	 * The recipients of this instance of the message. Typically this will be a single address.
	 */
	to: MailAddress[];
	/**
	 * The version of the MailerContent, currently only `1.0` is supported.
	 */
	version: string;
};

export function createContentBuffer(content: MailerContent): string {
	// fields are alphabetically ordered
	const canonicalized = canonicalize({
		authorMessagingKey: encodeHex(publicKeyToBytes(content.authorMessagingKey)),
		contentUri: content.contentUri,
		date: Math.round(content.date.getTime()),
		authorMailAddress: content.authorMailAddress.address,
		mailerProof: {
			params: {
				authorContentSignature: encodeHex(content.mailerProof.params.authorContentSignature),
				expires: Math.round(content.mailerProof.params.expires.getTime()),
				mailerMessagingKey: encodeHex(publicKeyToBytes(content.mailerProof.params.mailerMessagingKey)),
			},
			signature: encodeHex(content.mailerProof.signature),
			version: content.mailerProof.version,
		},
		messageId: content.messageId,
		to: content.to.map((recipient) => recipient.address),
		version: content.version,
	});

	if (!canonicalized) {
		throw new Error('content could not be canonicalized');
	}

	return canonicalized;
}

export function parseMailerContentFromJSON(content: string): MailerContent {
	type RawMailerContent = {
		authorMailAddress: string;
		authorMessagingKey: string;
		contentUri: string;
		date: number;
		mailerProof: {
			params: {
				expires: number;
				mailerMessagingKey: string;
				authorContentSignature: string;
			};
			signature: string;
			version: string;
		};
		messageId: string;
		to: string[];
		version: string;
	};

	const rawMailerContent: RawMailerContent = JSON.parse(content);

	if (!rawMailerContent.mailerProof) {
		throw new Error('mailerProof is required');
	}

	if (!rawMailerContent.mailerProof.params) {
		throw new Error('mailerProof.params is required');
	}

	if (rawMailerContent.mailerProof.params.authorContentSignature === '') {
		throw new Error('authorContentSignature is required');
	}

	const authorContentSignature = decodeHex(rawMailerContent.mailerProof.params.authorContentSignature);
	if (rawMailerContent.authorMessagingKey === '') {
		throw new Error('authorMessagingKey is required');
	}

	const authorMessagingKey = publicKeyFromBytes(decodeHex(rawMailerContent.authorMessagingKey));

	return {
		authorMessagingKey,
		contentUri: rawMailerContent.contentUri,
		date: new Date(rawMailerContent.date),
		authorMailAddress: { address: rawMailerContent.authorMailAddress, name: '' },
		to: rawMailerContent.to.map((address) => ({ address, name: '' })),
		mailerProof: {
			params: {
				expires: new Date(rawMailerContent.mailerProof.params.expires),
				mailerMessagingKey: publicKeyFromBytes(
					decodeHex(rawMailerContent.mailerProof.params.mailerMessagingKey),
				),
				authorContentSignature,
			},
			signature: decodeHex(rawMailerContent.mailerProof.signature),
			version: rawMailerContent.version,
		},
		messageId: rawMailerContent.messageId,
		version: rawMailerContent.version,
	};
}
