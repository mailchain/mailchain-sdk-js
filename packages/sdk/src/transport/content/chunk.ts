export const CHUNK_LENGTH_1MB = 1024 * 1024;

export function chunkBuffer(input: Uint8Array | Buffer, length: number): Buffer[] {
	let buffer = input instanceof Buffer ? input : Buffer.from(input);
	const result: Buffer[] = [];

	while (buffer.length > length) {
		const chunk = buffer.slice(0, length);
		buffer = buffer.slice(length);
		result.push(chunk);
	}

	if (buffer.length) {
		result.push(buffer);
	}

	return result;
}
