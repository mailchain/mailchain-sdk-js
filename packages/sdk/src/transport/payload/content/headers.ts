import { KindFromPublicKey, PublicKeyFromKind } from '@mailchain/crypto/multikey/names';
import { PublicKey } from '@mailchain/crypto/public';
import { DecodeBase64, EncodeBase64 } from '@mailchain/encoding';

type ContentType = 'application/json' | 'message/x.mailchain';
export type PayloadHeaders = {
	// Origin?: Uint8Array;
	/**
	 * public key of sender to verify the contents
	 */
	Origin: PublicKey;
	/**
	 * used to verify un-encrypted contents
	 */
	ContentSignature: Uint8Array;

	Created: Date;

	// Standard

	/**
	 * The size of the resource, in decimal number of bytes.
	 * The Content-Length header indicates the size of the message body, in bytes, sent to the recipient.
	 */
	ContentLength: number;

	/**
	 * Indicates the media type of the resource.
	 * The ContentType representation header is used to indicate the original media type of the resource (prior to any content encoding and encryption applied for sending).
	 * In responses, a ContentType header provides the client with the actual content type of the returned content.
	 */
	ContentType: ContentType;

	/**
	 * Used to specify the compression algorithm.
	 */
	ContentEncoding: string;

	/**
	 * Used to specify the encryption algorithm.
	 */
	ContentEncryption: string;

	/**
	 * Indicates an alternate location for the contents if not contained in this object.
	 */
	ContentLocation?: string;
};

const HEADER_CONTENT_ENCODING = 'Content-Encoding';
const HEADER_CONTENT_ENCRYPTION = 'Content-Encryption';
const HEADER_CONTENT_LENGTH = 'Content-Length';
const HEADER_CONTENT_LOCATION = 'Content-Location';
const HEADER_CONTENT_SIGNATURE = 'Content-Signature';
const HEADER_CONTENT_TYPE = 'Content-Type';
const HEADER_CREATED = 'Created';
const HEADER_ORIGIN = 'Origin';

export class SerializablePayloadHeaders {
	headers: PayloadHeaders;

	constructor(headers: PayloadHeaders) {
		this.headers = headers;
	}
	static FromEncryptedPayloadHeaders(headers: PayloadHeaders): SerializablePayloadHeaders {
		return new this(headers);
	}

	static FromBuffer(buffer: Buffer): SerializablePayloadHeaders {
		const invalidHeaders: string[] = [];
		const headers = new Map<String, String>();
		buffer
			.toString('utf8')
			.split('\r\n')
			.forEach((line) => {
				if (line.indexOf(':') === -1) {
					// splitting on ":" causes problem with dates
					invalidHeaders.push(line);
				}

				headers.set(line.slice(0, line.indexOf(':')).trim(), line.slice(line.indexOf(':') + 1).trim());
			});
		[
			HEADER_CONTENT_ENCODING,
			HEADER_CONTENT_ENCRYPTION,
			HEADER_CONTENT_LENGTH,
			HEADER_CONTENT_SIGNATURE,
			HEADER_CONTENT_TYPE,
			HEADER_CREATED,
			HEADER_ORIGIN,
		].forEach((item) => {
			if (!headers.get(item)) {
				throw new Error(`missing header '${item}'`);
			}
		});

		return new this({
			ContentEncoding: headers.get(HEADER_CONTENT_ENCODING)!.toString(),
			ContentEncryption: headers.get(HEADER_CONTENT_ENCRYPTION)!.toString(),
			ContentLength: Number.parseInt(headers.get(HEADER_CONTENT_LENGTH)!.toString()!),
			ContentSignature: parseSignatureHeader(headers.get(HEADER_CONTENT_SIGNATURE)!.toString()),
			ContentType: headers.get(HEADER_CONTENT_TYPE)!.toString() as ContentType,
			Created: new Date(headers.get(HEADER_CREATED)!.toString()!),
			Origin: parseOriginHeader(headers.get(HEADER_ORIGIN)!.toString()),
		});
	}

	ToBuffer(): Buffer {
		const headers: string[] = [];
		headers.push(`${HEADER_CONTENT_ENCODING}: ${this.headers.ContentEncoding}`);
		headers.push(`${HEADER_CONTENT_ENCRYPTION}: ${this.headers.ContentEncryption}`);
		headers.push(`${HEADER_CONTENT_LENGTH}: ${this.headers.ContentLength}`);
		if (this.headers.ContentLocation) {
			headers.push(`${HEADER_CONTENT_LOCATION}: ${this.headers.ContentLocation}`);
		}
		headers.push(`Content-Signature: ${createSignatureHeader(this.headers.ContentSignature, this.headers.Origin)}`);
		headers.push(`Content-Type: ${this.headers.ContentType}`);
		headers.push(`Created: ${this.headers.Created.toISOString()}`);

		headers.push(`Origin: ${createOriginHeader(this.headers.Origin)}`);

		return Buffer.from(headers.join('\r\n'), 'utf8');
	}
}

function parseHeaderElements(input: string, requiredKeys: string[]): Map<String, String> {
	const invalidAttributes: string[] = [];
	const attributes = new Map<String, String>();
	input.split(';').forEach((item) => {
		const parts = item.split('=', 2);
		if (parts.length !== 2) {
			invalidAttributes.push(item);
		}
		attributes.set(parts[0].trim(), parts[1].trim());
	});

	requiredKeys.forEach((item) => {
		if (!attributes.get(item)) {
			throw new Error(`missing header attribute '${item}'`);
		}
	});

	return attributes;
}

function parseSignatureHeader(input: string): Uint8Array {
	const attributes = parseHeaderElements(input, ['data']);

	const sig = DecodeBase64(attributes.get('data')!.toString());

	if (sig.length === 0) {
		throw new Error('could not decode signature');
	}

	return sig;
}

function createSignatureHeader(signature: Uint8Array, signer: PublicKey): string {
	const values: string[] = [];
	values.push(`data=${EncodeBase64(signature)}`);
	values.push(`alg=${KindFromPublicKey(signer)}`);

	return values.join('; ').trimEnd();
}

function parseOriginHeader(input: string): PublicKey {
	const attributes = parseHeaderElements(input, ['data', 'alg']);
	const bytes = DecodeBase64(attributes.get('data')!.toString());

	return PublicKeyFromKind(attributes.get('alg')!.toString(), bytes);
}

function createOriginHeader(signer: PublicKey): string {
	const values: string[] = [];
	values.push(`data=${EncodeBase64(signer.bytes)}`);
	values.push(`alg=${KindFromPublicKey(signer)}`);

	return values.join('; ').trimEnd();
}
