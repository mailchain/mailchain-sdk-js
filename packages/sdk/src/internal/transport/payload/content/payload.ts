import { PayloadHeaders } from './headers';

/**
 * Payload.
 */
export interface Payload {
	/**
	 * @see PayloadHeaders
	 */
	Headers: PayloadHeaders;
	/**
	 * Raw/data/object that are decrypted and parsed base on the the headers.
	 */
	Content: Buffer;
}

/**
 * Encrypted payload.
 */
export interface EncryptedPayload {
	EncryptedHeaders: Buffer;

	EncryptedContentChunks: Buffer[];
}
