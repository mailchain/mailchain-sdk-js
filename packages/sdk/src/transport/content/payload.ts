import { PayloadHeaders } from './headers';

/**
 * Payload.
 */
export interface Payload {
	Headers: PayloadHeaders;
	/**
    //  * Raw/data/object that are decrypted and parsed base on the the headers.
     */
	Content: Buffer;
}

/**
 * Encrypted payload.
 */
export interface EncryptedPayload {
	EncryptedHeaders: Buffer;

	EncryptedContentChunks: Buffer[];
}
