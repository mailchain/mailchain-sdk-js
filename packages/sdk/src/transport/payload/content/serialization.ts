import { EncryptedPayload } from './payload';

export enum MessageKind {
	UNSET = 0,
	HEADER = 1,
	CHUNK = 2,
	FINAL_CHUNK = 3,
}

enum Version {
	UNSET = 0,
	/**
	 * Uses 32 byte numbers for size of messages
	 */
	V1_0_0 = 1,
}

const SIZE_VERSION = 1;
const SIZE_MESSAGE_KIND = 1;
const SIZE_MESSAGE_BYTES_SIZE = 4;

export function serializeMessage(encryptedMessage: Buffer, messageKind: MessageKind): Buffer {
	if (messageKind === MessageKind.UNSET) {
		throw Error('message kind invalid');
	}
	const encryptedMessageSize = SIZE_MESSAGE_KIND + SIZE_MESSAGE_BYTES_SIZE + encryptedMessage.length;
	const metadata = Buffer.alloc(SIZE_MESSAGE_KIND + SIZE_MESSAGE_BYTES_SIZE);
	const endPositionKind = metadata.writeUInt8(messageKind, 0);
	metadata.writeUInt32BE(encryptedMessage.length, endPositionKind);

	return Buffer.concat([metadata, encryptedMessage], encryptedMessageSize);
}

export function Serialize(input: EncryptedPayload): Buffer {
	const bufferCollector = new Array<Buffer>(input.EncryptedContentChunks.length + 1 + 1);

	bufferCollector[0] = Buffer.from(new Uint8Array([Version.V1_0_0]));
	let totalBufferOutSize = SIZE_VERSION;

	const serializedHeader = serializeMessage(input.EncryptedHeaders, MessageKind.HEADER);
	bufferCollector[1] = serializedHeader;
	totalBufferOutSize += serializedHeader.length;

	input.EncryptedContentChunks.forEach((c, i) => {
		const kind = input.EncryptedContentChunks.length === i + 1 ? MessageKind.FINAL_CHUNK : MessageKind.CHUNK;
		const serializedChunk = serializeMessage(c, kind);
		bufferCollector[i + 2] = serializedChunk;
		totalBufferOutSize += serializedChunk.length;
	});

	return Buffer.concat(bufferCollector, totalBufferOutSize);
}

export function deserializeMessage(serializedMessage: Buffer): {
	kind: MessageKind;
	message: Buffer;
	remainingBuffer: Buffer;
} {
	if (serializedMessage.length < SIZE_MESSAGE_KIND + SIZE_MESSAGE_BYTES_SIZE) {
		throw new Error('message segment too short to open');
	}

	if (!(serializedMessage[0] in MessageKind)) {
		throw new Error('message kind invalid');
	}
	const kind = serializedMessage[0] as MessageKind;
	if (kind === MessageKind.UNSET) {
		throw new Error('message kind UNSET');
	}

	const messageSize = Number(serializedMessage.readUInt32BE(SIZE_MESSAGE_KIND));

	if (serializedMessage.length < messageSize + SIZE_MESSAGE_KIND + SIZE_MESSAGE_BYTES_SIZE) {
		throw new Error('message segment has missing data');
	}

	const offset = SIZE_MESSAGE_KIND + SIZE_MESSAGE_BYTES_SIZE;
	const end = offset + messageSize;
	const remainingBuffer = serializedMessage.slice(end);

	return {
		kind,
		message: serializedMessage.slice(offset, end),
		remainingBuffer,
	};
}

export function Deserialize(serializedData: Buffer): EncryptedPayload {
	const version = serializedData[0] as Version;
	if (version !== Version.V1_0_0) {
		throw new Error('serialization version not supported');
	}

	const headerPart = deserializeMessage(serializedData.slice(1));
	if (headerPart.kind !== MessageKind.HEADER) {
		throw new Error('expected header part');
	}

	let { remainingBuffer } = headerPart;
	let lastMessageKind = headerPart.kind as MessageKind;
	const deserializedChunks = new Array<Buffer>();

	while (remainingBuffer.length > 0) {
		const deserializedChunk = deserializeMessage(remainingBuffer);
		if (deserializedChunk.kind !== MessageKind.CHUNK && deserializedChunk.kind !== MessageKind.FINAL_CHUNK) {
			throw new Error('expected chunk variant');
		}
		deserializedChunks.push(deserializedChunk.message);

		remainingBuffer = deserializedChunk.remainingBuffer;
		lastMessageKind = deserializedChunk.kind;
	}

	// zero chunks is possible due to content location header
	if (deserializedChunks.length > 0 && lastMessageKind !== MessageKind.FINAL_CHUNK) {
		throw new Error('last chunk must be terminating chunk');
	}

	return {
		EncryptedContentChunks: deserializedChunks,
		EncryptedHeaders: headerPart.message,
	};
}
