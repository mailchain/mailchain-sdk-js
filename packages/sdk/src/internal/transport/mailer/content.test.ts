import { AliceED25519PublicKey, BobED25519PublicKey } from '@mailchain/crypto/ed25519/test.const';
import { createContentBuffer, MailerContent, parseMailerContentFromJSON } from './content';

describe('createContentBuffer', () => {
	it('example payload', () => {
		const actual = createContentBuffer({
			version: '1',
			contentUri: 'https://example.com',
			contentHash: new Uint8Array([0x01, 0x02, 0x03, 0x04]),
			date: new Date('2020-06-06'),
			authorMailAddress: {
				name: 'Alice',
				address: 'alice@mailchain.com',
			},
			to: [
				{
					name: 'Bob',
					address: 'bob@mailchain.com',
				},
			],
			mailerProof: {
				params: {
					expires: new Date('2022-06-06'),
					mailerMessagingKey: BobED25519PublicKey,
					authorContentSignature: new Uint8Array([0x05, 0x06, 0x07, 0x08]),
				},
				signature: new Uint8Array([0x0a, 0xb, 0xc, 0xd]),
				version: '1.0',
			},
			messageId: 'message-id-0',
			authorMessagingKey: AliceED25519PublicKey,
			authorContentSignature: new Uint8Array([0x09, 0x10, 0x11, 0x12]),
		} as MailerContent);

		expect(actual).toMatchSnapshot('original');
	});
});

describe('round trip', () => {
	it('example payload', () => {
		const original: MailerContent = {
			version: '1.0',
			contentUri: 'https://example.com',
			date: new Date('2023-01-18T18:10:38.896Z'),
			authorMailAddress: {
				name: '',
				address: 'alice@mailchain.com',
			},
			to: [
				{
					name: '',
					address: 'bob@mailchain.com',
				},
			],
			mailerProof: {
				params: {
					expires: new Date('2022-06-06'),
					mailerMessagingKey: BobED25519PublicKey,
					authorContentSignature: new Uint8Array([0x05, 0x06, 0x07, 0x08]),
				},
				signature: new Uint8Array([0x0a, 0xb, 0xc, 0xd]),
				version: '1.0',
			},
			messageId: 'message-id-0',
			authorMessagingKey: AliceED25519PublicKey,
		};

		const bufferedContent = createContentBuffer(original);

		const actual = parseMailerContentFromJSON(bufferedContent);

		expect(actual).toEqual(original);
	});
});
