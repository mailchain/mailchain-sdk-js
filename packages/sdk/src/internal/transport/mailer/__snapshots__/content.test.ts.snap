// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createContentBuffer example payload: original 1`] = `"{"authorMailAddress":"alice@mailchain.com","authorMessagingKey":"e2723caa23a5b511af5ad7b7ef6076e414ab7e75a9dc910ea60e417a2b770a5671","contentUri":"https://example.com","date":1591401600000,"mailerProof":{"params":{"authorContentSignature":"05060708","expires":1654473600000,"mailerMessagingKey":"e22e322f8740c60172111ac8eadcdda2512f90d06d0e503ef189979a159bece1e8"},"signature":"0a0b0c0d"},"messageId":"message-id-0","to":["bob@mailchain.com"],"version":"1"}"`;
